lib/src/precompute_client.dart

```diff
@@ -0,0 +1,595 @@
+      flagKey,
+      defaultValue,
+      VariationType.string,
+    );
+  }
+
+  /// Gets a boolean assignment for a flag
+  bool getBooleanAssignment(String flagKey, bool defaultValue) {
+    return _getPrecomputedAssignment(
+      flagKey,
+      defaultValue,
+      VariationType.boolean,
+    );
+  }
+
+  /// Gets an integer assignment for a flag
+  int getIntegerAssignment(String flagKey, int defaultValue) {
+    return _getPrecomputedAssignment(
+      flagKey,
+      defaultValue,
+      VariationType.integer,
+    );
+  }
+
+  /// Gets a numeric (double) assignment for a flag
+  double getNumericAssignment(String flagKey, double defaultValue) {
+    return _getPrecomputedAssignment(
+      flagKey,
+      defaultValue,
+      VariationType.numeric,
+    );
+  }
+
+  /// Gets a JSON object assignment for a flag
+  Map<String, dynamic> getJSONAssignment(
+    String flagKey,
+    Map<String, dynamic> defaultValue,
+  ) {
+    return _getPrecomputedAssignment(
+      flagKey,
+      defaultValue,
+      VariationType.json,
+      (value) {
+        if (value is String) {
+          try {
+            return JsonDecoder().convert(value) as Map<String, dynamic>;
+          } catch (e) {
+            _logger.warning('$defaultLoggerPrefix Error parsing JSON: $e');
+            return defaultValue;
+          }
+        } else if (value is Map) {
+          // If the value is already a Map (decoded from JSON), convert it to Map<String, dynamic>
+          try {
+            return Map<String, dynamic>.from(value);
+          } catch (e) {
+            _logger.warning('$defaultLoggerPrefix Error converting Map: $e');
+            return defaultValue;
+          }
+        }
+        return defaultValue;
+      },
+    );
+  }
+
+  /// Gets a bandit action for a flag
+  BanditEvaluation getBanditAction(
+    String flagKey,
+    String defaultValue,
+  ) {
+    final obfuscatedBandit = _getPrecomputedBandit(flagKey);
+
+    if (obfuscatedBandit == null) {
+      _logger.warning(
+        '$defaultLoggerPrefix No assigned variation. Bandit not found: $flagKey',
+      );
+      return BanditEvaluation(variation: defaultValue, action: null);
+    }
+
+    // Decode the bandit inline
+    final decodedBanditKey = decodeBase64(obfuscatedBandit.banditKey);
+    final decodedAction = decodeBase64(obfuscatedBandit.action);
+    final decodedModelVersion = decodeBase64(obfuscatedBandit.modelVersion);
+    final decodedActionNumericAttributes =
+        obfuscatedBandit.actionNumericAttributes.map(
+      (key, value) =>
+          MapEntry(key, double.tryParse(decodeBase64(value)) ?? 0.0),
+    );
+    final decodedActionCategoricalAttributes =
+        obfuscatedBandit.actionCategoricalAttributes.map(
+      (key, value) => MapEntry(key, decodeBase64(value)),
+    );
+
+    final assignedVariation = getStringAssignment(flagKey, defaultValue);
+
+    final banditEvent = BanditEvent(
+      timestamp: DateTime.now().toIso8601String(),
```

Comment:
Thinking if we do stick with strings, we'll want to call `.toUtc()` when we do this so we don't have local times all over

Reply:
Switched to `DateTime.timestamp()`

```
/// Constructs a [DateTime] with the current UTC date and time.
  ///
  ///
  /// ```dart
  /// final mark = DateTime.timestamp();
  /// ```
  @Since("3.0")
  DateTime.timestamp() : this._nowUtc();
```